include/nlohmann/json.hpp

```diff
@@ -1234,7 +1238,8 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
     /// @brief move constructor
     /// @sa https://json.nlohmann.me/api/basic_json/basic_json/
     basic_json(basic_json&& other) noexcept
-        : m_type(std::move(other.m_type)),
+        : json_base_class_t(static_cast < json_base_class_t&& > (other)),
```

Comment:
Please keep `std::move` and use `NOLINT` instead.
```cpp
: json_base_class_t(std::move(other)), // NOLINT(misc-use-after-move)
```

Reply:
Done